### PR TITLE
[1/3] Delivery Tracking: Refactor UserDefaults usage

### PR DESCRIPTION
**Pull request by ftcvlad**
### Description of Changes

1) Move access to UserDefaults into a helper class.
2) Remove deprecated `synchronize`
3) use `UserDefaults.standard.object` instead of `UserDefaults.standard.value`

### Breaking Changes

-   None

### Release Checklist

Prepare:

-   [ ] Detail any breaking changes. Breaking changes require a new major version number
-   [ ] Check `pod lib lint` passes

Bump versions in:

-   [ ] `Sources/Kumulos.swift`
-   [ ] `KumulosSdkSwift.podspec`
-   [ ] `KumulosSdkSwiftExtension.podspec`
-   [ ] `Build target version for plist`
-   [ ] `README.md`

Release:

-   [ ] Squash and merge to master
-   [ ] Delete branch once merged
-   [ ] Create tag from master matching chosen version
-   [ ] Run `pod trunk push` to publish to CocoaPods

### CODE REVIEW

File: Sources/InApp/InAppHelper.swift [hunk 1]
```diff
@@ -149,7 +149,7 @@ internal class InAppHelper {
     }
     
     func userConsented() -> Bool {
-        return UserDefaults.standard.bool(forKey: KUMULOS_IN_APP_CONSENTED_KEY);
+        return PersistenceHelper.object(forKey: KUMULOS_IN_APP_CONSENTED_KEY) as! Bool;
```
Comment (cgwyllie) [inline Sources/InApp/InAppHelper.swift]: Not 100% sure this is equivalent? https://developer.apple.com/documentation/foundation/userdefaults/1416388-bool

Will force-unwrapping nil to Bool have an issue?

File: PersistenceHelper.swift [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+//  PersistenceHelper.swift
+//  KumulosSDK
+//
+//  Created by Vladislav Voicehovics on 13/03/2020.
+//  Copyright © 2020 Kumulos. All rights reserved.
+//
+
+import Foundation
+
+internal class PersistenceHelper {
```
Comment (cgwyllie) [inline PersistenceHelper.swift]: Thought on naming (if this stays as a k-v abstraction), maybe KeyValPersistenceHelper.. or similar?

File: Sources/InApp/InAppHelper.swift (hunk 1 above)
Comment (ftcvlad) [reply Sources/InApp/InAppHelper.swift]: 🤦‍♂

File: PersistenceHelper.swift (hunk 2 above)
Comment (ftcvlad) [reply PersistenceHelper.swift]: k

**Pull request merged**